### PR TITLE
add functionality to work with locally compiled liboboe

### PR DESCRIPTION
**Pull request by prettycoder**
- separate extconf (to make sure there is no confusion)
- additional dir in docker container for compiled oboe
- OBOE_LOCAL env var to indicate that locally compiled liboboe should be linked
- changes to test script

### CODE REVIEW

File: ext/oboe_metal/extconf_local.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2016 SolarWinds, LLC.
```
Comment (ronilan) [inline ext/oboe_metal/extconf_local.rb:3]: I see this on all files (with 2016). There is none such for Node and other repos. Not sure it is legally needed these days.

File: Rakefile [hunk 2]
```diff
@@ -184,27 +184,29 @@ task :fetch_ext_deps do
...
-        content = rf.read
-        File.open(local_file, 'wb') { |f| f.puts content }
-        puts "%%% #{filename} checksum: #{content.strip} %%%"
-      end
-    else
-      URI.open(remote_file, 'rb') do |rf|
-        content = rf.read
-        File.open(local_file, 'wb') { |f| f.puts content }
-        puts "%%% #{filename} checksum: #{content.strip} %%%"
+  unless ENV['OBOE_LOCAL']
```
Comment (ronilan) [inline Rakefile:187]: `unless` is a such a ruby beauty....

**Review by ronilan — COMMENTED** [review]
It is a pleasure reading ruby code and all changes LGTM.

There are however two GH Actions failures in tests, both for ruby 2.5 and pg and unrelated to code changes in PR. Given possible "flakiness", and current inability to rerun just one GH Actions job in matrix, and in attempt to avoid distractions, me think it might be better to not run "whole enchilada" for `on: pull_request`.

File: Rakefile (hunk 2 above)
Comment (prettycoder) [reply Rakefile:187]: some people hate it ;)

File: ext/oboe_metal/extconf_local.rb (hunk 1 above)
Comment (prettycoder) [reply ext/oboe_metal/extconf_local.rb:3]: it is recommended practice
the year should be the year the code was first written, it does not need to be updated
we have it in the python agent as well

**Review by ronilan — APPROVED** [review]

**Pull request merged**